desktop-widgets/subsurfacewebservices.cpp

```diff
@@ -110,7 +110,7 @@ static bool merge_locations_into_dives(void)
 								break;
 							} else {
 								/* ok, gpsfix is before, nextgpsfix is after */
-								if (dive->when - gpsfix->when <= nextgpsfix->when - dive_endtime(dive)) {
+								if (dive->when - gpsfix->when <= nextgpsfix->when - dive->when) {
```

Comment:
Again, this makes assumptions about the dive style.
In the case where you have a boat that anchors and you dive from there, I would claim that you are wrong.

Reply:
Didn't think of this point, just thought it was a failure. Agree with you.